scripts/convert_config.py

```diff
@@ -0,0 +1,460 @@
+
+            spec.loader.exec_module(module)
+
+            # Run adjust function from adjust script
+            execute = getattr(module, "adjust")
+            execute(new, mapping)
+        except Exception as e:
+            print(f"Failed while running adjustment from {adjust_script}: {e}")
+
+    if debug:
+        print(json.dumps(new, indent=4))
+
+    return new
+
+
+def is_version(version_string):
+    """
+    Validate a string containing a version number
+
+    return: True if the string is a valid version number; False otherwise
+    """
+
+    if not version_string:
+        raise Exception("String not provided")
+
+    # Remove any leading or trailing white spaces
+    version_string = version_string.strip()
+
+    regex = re.compile(r'^\d+\.\d+$')
+
+    if regex.match(version_string):
+        return True
+    else:
+        return False
+
+
+def cmp_version(a, b):
+    """
+    Compare two version strings and return 0 if both are equal, a negative
+    number if the second is greater than the first and a positive number if the
+    first is greater than the second
+    """
+
+    if not is_version(a):
+        raise Exception(f"Invalid version string {a}: expected 'MAJOR.MINOR'")
+
+    if not is_version(b):
+        raise Exception(f"Invalid version string {b}: expected 'MAJOR.MINOR'")
+
+    # After validating the strings format, split and compare
+    (a_major, a_minor) = map(lambda n: int(n), a.strip().split('.'))
+    (b_major, b_minor) = map(lambda n: int(n), b.strip().split('.'))
+
+    if a_major == b_major:
+        return a_minor - b_minor
+    else:
+        return a_major - b_major
+
+
+def process_versions(templates, old_config, target_version=None, debug=False):
+    """
+    Apply the transformations from the mappings for each version found in the
+    templates folder to the configuration.
+
+    If a target version is provided, the process will stop if the target version
+    is reached, otherwise all transformations to the latest version are applied
+    """
+
+    versions = os.listdir(templates)
+
+    if not versions:
+        raise Exception(f"No directories found in {templates} for versions")
+
+    # Remove from the list all directories that are not named as a version
+    # number
+    versions = filter(is_version, versions)
+
+    # Get the ordered list of available version numbers
+    versions = sorted(os.listdir(templates),
+                      key=cmp_to_key(cmp_version))
+
+    if target_version and not target_version in versions:
+        raise Exception(f"Directory for target version {target_version} not "
+                        f"found in {templates}")
+
+    new = {}
+
+    # If the configuration file does not contain a version number, assume it is
+    # in the old format and use the minimum possible version
+    if "version" in old_config.keys():
+        old_version = old_config["version"]
+    else:
+        old_version = '0.0'
```

Comment:
All can be replaced with `old_version = old_config.get("version", "0.0")`

Reply:
Also, should be `"1.0"`?

Reply:
Thanks, I'll replace everything with `old_version = old_config.get("version", "1.0")`

Since we changed the first version in templates to be `2.0`, we can use `1.0` here.

Reply:
Using `old_version = old_config.get("version", "1.0")` was not straightforward as I initially thought since `old_config` is not always a normal python dictionary, but can also be a `RawConfigParser` which implements `get` differently. I worked around by catching the exceptions that can be raised. Please check if this is acceptable.